diff --git a/RgbaInput.tsx b/RgbaInput.tsx
index e7309bf..2995b9a 100644
--- a/RgbaInput.tsx
+++ b/RgbaInput.tsx
@@ -1,20 +1,19 @@
-import React from 'react';
+import React, { useRef } from 'react';
 
 export const RgbaInput = ({ rgba, setRgba }) => {
-  const onChange = idx => ({ target: { value } }) => {
-    const newRgba = [...rgba];
-    newRgba[idx] = value;
+  const inputs = [useRef(null), useRef(null), useRef(null), useRef(null)];
+  const onChange = () => {
+    const colorValues = inputs.map(({ current: { value } }) => value);
 
-    setRgba(newRgba);
+    setRgba(colorValues);
   };
 
   return (
     <label>
       RGB(A):
-      <input value={rgba[0]} onChange={onChange(0)} />
-      <input value={rgba[1]} onChange={onChange(1)} />
-      <input value={rgba[2]} onChange={onChange(2)} />
-      <input value={rgba[3]} onChange={onChange(3)} />
+      {inputs.map((input, idx) => (
+        <input onChange={onChange} key={'rgba ' + idx} ref={input} />
+      ))}
     </label>
   );
-}
+};
